File: services/backend/src/core_service/modules/publications/domain/publication.py
"""Publication 도메인 엔티티 — schema.md §3.17 `publications` 테이블 매핑.

`chapters`/`chapter_revisions`와 달리 이 엔티티 자체는 PII 자유텍스트 컬럼이 없다
(완성본은 MinIO 오브젝트로 저장되고 `storage_ref`는 내부 키일 뿐). 원문(챕터 body_text)은
`BookBuilderService`가 조판 시점에만 잠깐 복호화해서 쓰고 이 테이블엔 남기지 않는다.
"""

from dataclasses import dataclass
from datetime import datetime
from enum import StrEnum
from uuid import UUID


class PublicationFormat(StrEnum):
    HARDCOVER_PDF = "hardcover_pdf"
    EPUB = "epub"


class PublicationStatus(StrEnum):
    REQUESTED = "requested"
    PROCESSING = "processing"
    READY = "ready"
    DELIVERED = "delivered"


@dataclass
class Publication:
    id: UUID
    user_id: UUID
    format: PublicationFormat
    status: PublicationStatus
    storage_ref: str | None
    requested_at: datetime
    completed_at: datetime | None

File: apps/web/src/types/publication.ts
/** design.md §3.1/§2.6 Publication — schema.md §3.17 `publications` 매핑.
 * `services/backend`의 publications API(publications/api/v1/publications.py
 * PublicationResponse)와 1:1. */
export type PublicationFormat = "hardcover_pdf" | "epub";
export type PublicationStatus = "requested" | "processing" | "ready" | "delivered";

export interface Publication {
  id: string;
  userId: string;
  format: PublicationFormat;
  status: PublicationStatus;
  requestedAt: string;
  completedAt?: string;
  /** MinIO presigned GET URL, 15분 만료 — status가 ready/delivered일 때만 존재. */
  downloadUrl: string | null;
}

export const PUBLICATION_FORMAT_LABEL: Record<PublicationFormat, string> = {
  hardcover_pdf: "하드커버 PDF",
  epub: "전자책(ePub)",
};

export const PUBLICATION_STATUS_LABEL: Record<PublicationStatus, string> = {
  requested: "접수됨",
  processing: "제작 중",
  ready: "완성됨",
  delivered: "전달 완료",
};
